src/integrationTest/java/uk/gov/hmcts/reform/notificationservice/data/NotificationRepositoryTest.java

```diff
@@ -55,6 +58,185 @@ void should_return_empty_optional_when_there_is_no_notification_in_db() {
         assertThat(notificationRepository.find(1_000)).isEmpty();
     }
 
+    @Test
+    void should_find_single_notification_by_zip_file_name_and_service() {
+        // given
+        final var zipFileName = "zip_file_name";
+        final var service = "service";
+
+        final var newNotification = new NewNotification(
+            zipFileName,
+            "po_box",
+            service,
+            "dcn",
+            ErrorCode.ERR_AV_FAILED,
+            "error_description"
+        );
+
+        // when
+        notificationRepository.insert(newNotification);
+
+        // and
+        List<Notification> notifications = notificationRepository.find(zipFileName, service);
+
+        // then
+        assertThat(notifications)
+            .isNotEmpty()
+            .hasSize(1)
+            .extracting(this::getTupleFromNotification)
+            .containsExactlyInAnyOrder(
+                tuple(
+                    zipFileName,
+                    newNotification.poBox,
+                    service,
+                    newNotification.documentControlNumber,
+                    newNotification.errorCode,
+                    newNotification.errorDescription,
+                    null,
+                    PENDING
+                )
```

Comment:
it's only one object, can we explicitly check fields instead (for clarity what's what)